eras/conway/impl/src/Cardano/Ledger/Conway/Rules/GovCert.hs

```diff
@@ -264,11 +265,14 @@ conwayGovCertTransition = do
                 vsDReps
           }
     ConwayUnRegDRep cred refund -> do
-      case Map.lookup cred vsDReps of
-        Nothing -> failBecause $ ConwayDRepNotRegistered cred
-        Just drepState ->
-          let paidDeposit = drepState ^. drepDepositL
-           in refund == paidDeposit ?! ConwayDRepIncorrectRefund refund paidDeposit
+      let mDRepState = Map.lookup cred vsDReps
+          drepRefundMismatch = do
+            drepState <- mDRepState
+            let paidDeposit = drepState ^. drepDepositL
+            guard (refund /= paidDeposit)
+            pure paidDeposit
+      isJust mDRepState ?! ConwayDRepNotRegistered cred
```

Comment:
Probably because it's implementation would be `failOnNothing x = (isJust x ?!)`

However, just for consistency sake with `failOnJust` we could add it.